modules/cpmso-yaml-provider-spec-azure.adoc

```diff
@@ -0,0 +1,69 @@
+// Module included in the following assemblies:
+//
+// * machine_management/control-plane-machine-management.adoc
+
+:_content-type: REFERENCE
+[id="cpmso-yaml-provider-spec-azure_{context}"]
+= Sample Azure provider specification
+
+When you create a control plane machine set for an existing cluster, the provider specification must match the `providerSpec` configuration in the control plane `Machine` CR that is created by the installation program. You can omit any field that is set in the failure domain section of the CR.
+
+In the following example, `<cluster_id>` is the infrastructure ID that is based on the cluster ID that you set when you provisioned the cluster. If you have the OpenShift CLI installed, you can obtain the infrastructure ID by running the following command:
+
+[source,terminal]
+----
+$ oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster
+----
+
+.Sample Azure `providerSpec` values
+[source,yaml]
+----
+providerSpec:
+  value:
+    acceleratedNetworking: true
+    apiVersion: machine.openshift.io/v1beta1
+    credentialsSecret:
+      name: azure-cloud-credentials <1>
+      namespace: openshift-machine-api
+    diagnostics: {}
+    image: <2>
+      offer: ""
+      publisher: ""
+      resourceID: /resourceGroups/<cluster_id>-rg/providers/Microsoft.Compute/galleries/gallery_<cluster_id>/images/<cluster_id>-gen2/versions/412.86.20220930 <3>
+      sku: ""
+      version: ""
+    internalLoadBalancer: <cluster_id>-internal <4>
+    kind: AzureMachineProviderSpec <5>
+    location: <region> <6>
+    managedIdentity: <cluster_id>-identity
+    metadata:
+      creationTimestamp: null
+      name: <cluster_id>
+    networkResourceGroup: <cluster_id>-rg
+    osDisk: <7>
+      diskSettings: {}
+      diskSizeGB: 1024
+      managedDisk:
+        storageAccountType: Premium_LRS
+      osType: Linux
+    publicIP: false
+    publicLoadBalancer: <cluster_id> <8>
+    resourceGroup: <cluster_id>-rg
+    subnet: <cluster_id>-master-subnet <9>
+    userDataSecret:
+      name: master-user-data <10>
+    vmSize: Standard_D8s_v3
+    vnet: <cluster_id>-vnet
+    zone: "" <11>
```

Comment:
should remove this, this is defined in the failureDomains part.

Reply:
Talking to Joel, my understanding was that if this value is present, it gets overwritten by the value in the `failureDomains` section. That callout `<11>` maps to "This parameter is configured in the failure domain, and is shown with an empty value here. If a value specified for this parameter differs from the value in the failure domain, the Operator overwrites it with the value in the failure domain." I was trying to show that this is where it might go, but that the user should refer to the `failureDomains` stanza instead.

Do you think this explanation makes sense?

Reply:
yes @jeana-redhat that makes sense, thank you.